### ISSUE TITLE
macOS Monterey support

### ISSUE THREAD

**Issue by patiman** [ExistentialAudio/BlackHole#500]
Hi,

When will you release updated version of BlackHole compatible with macOS Monterey version 12.0.1 on Apple M1 chip?

Thank you

**Comment by devinroth** [comment]
Tested it and it works fine for me using that exact same configuration. 

That being said I am working on an update. Please give this beta version a run see if it works any better. 
[BlackHole2ch.v0.2.11-beta.pkg.zip](https://github.com/ExistentialAudio/BlackHole/files/7623104/BlackHole2ch.v0.2.11-beta.pkg.zip)

**Comment by yputter** [comment]
I also have a MacBook Air M1 running on macOS Montery version 12.0.1, but doesn't seem to work. 

In my specific case, I'm trying to send both my piano app (Addictive Keys) output and MacBook Mic as input to Skype, as well as hearing it myself on my headphones.

In Audio MIDI Setup
- Aggregated Device, with first MacBook Air Microphone (can't select a Built-in thingie) and then Blackhole ch2.
- Multi-Output Device, with first the External Headphone (can't select a Built-in thingie) and then Blackhole ch2.

In Addictive Keys
- I routed output to Multi-Output Device

In Skype
- select Aggregated Device as input
- select Multi-Output Device as speakers

**Comment by LOpSSH** [comment]
> Tested it and it works fine for me using that exact same configuration.
> 
> That being said I am working on an update. Please give this beta version a run see if it works any better. [BlackHole2ch.v0.2.11-beta.pkg.zip](https://github.com/ExistentialAudio/BlackHole/files/7623104/BlackHole2ch.v0.2.11-beta.pkg.zip)

doesn't seem to work for me.

**Comment by MarqueIV** [comment]
I'm running Montery 12.0.1 on an Intel MacBook Pro and am trying to record my notebooks output and I can't get this to work either.

When I create a multi-output device, and set 'Built In Output' as the topmost output, followed by 'BlackHole 2ch', while I can hear the output through the speakers, choosing 'BlackHole 2ch' in my recording software shows zero levels for the audio and nothing gets recorded.

If I change the output directly to 'BlackHole 2ch', then I *can* record the system's sounds, but of course I can't then hear them since all output is being routed to 'BlackHole 2ch' and not the actual speakers.

I also tried installing the beta above here and the installation fails with the following message...

<img width="732" alt="image" src="https://user-images.githubusercontent.com/897883/144726017-162aa8cc-32bd-452a-9e8d-d6df0ff5764d.png">

**Comment by JP0047** [comment]
um I'm new here and I don't know if this is the right place to post this but I tried searching the internet for an answer but I haven't been able to find what I was looking for
 
I have an M1 mac and when I open the Audio MIDI Setup from utilities it shows that blackholes input and output sources are set to default while it should show the individual source for Eg: For Blackhole the source should be Blackhole and so forth but over here it doesn't work for some reason

I'd really appreciate it if someone could explain to me if this is normal and if it will disrupt the way I route audio from my DAW to Zoom using the Multi-Output device

<img width="638" alt="Screen Shot 2021-12-06 at 6 48 39 PM" src="https://user-images.githubusercontent.com/95634130/144867882-c9abb1f7-0f29-4218-b8e7-84354d425a6a.png">

**Comment by baodad** [comment]
Same here.  Cannot get Blackhole to work as loopback audio interface on M1 mac with Monterey 12.0.1.  I tried both 2ch and 16ch versions of Blackhole.

Cheers for all your work.  Hope this feedback is helpful.
Also tried downloading the beta version shared above, but the installation failed despite running installer disregarding virus warning.

**Comment by molk-ibm** [comment]
> I also tried installing the beta above here and the installation fails with the following message...

Same here on Monterey. Can't install the Beta. Installation of latest works, but can't select for output.

<img width="640" alt="grafik" src="https://user-images.githubusercontent.com/90783703/147766460-c469a61b-1d74-42bc-962c-02c241ddb6bd.png">

**Comment by seenickcode** [comment]
Unable to install the beta 0.2.11 on Monterey. Installation fails. 

Still getting very distorted audio for 2ch and 16ch versions.

**Comment by harriska2** [comment]
I'm trying to get the 2 channel version working on my intel Mac Mini running Monterey and no joy. It installs. I set input to blackHole, out put to either hdmi or combined blackhole & hdmi and neither picks up sound from a screen capture video.

**Comment by seenickcode** [comment]
I just ended up using Loopback.app. Had to pay $99 but heck, it works.

**Comment by harriska2** [comment]
I installed Soundflower after BlackHole. It required a reboot. SoundFlower didn’t work but BlackHole started working.

**Comment by ubaumgar** [comment]
It seems as if it would need some time to bring back BlackHole fully functioning to macOS Monterey. Here is what I did to continue working in the mean time. I am using OBS+ and need BlackHole to route the monitoring audio to a virtual mic for using it with Zoom or others.
There is obviously other software such as Loopback available but for quite some cost and I personally liked working with BlackHole and would like to stick with it. That's why I was looking for cheaper alternatives and came up with two solutions (both with hardware though) and a cost of about 10$-30$:

Alternative 1: 
A soundcard connected to the USB and a simple 3-pol cable that just connects the sound output and the sound input on the card. I use a Sandberg Sound Link (https://sandberg.world/en-ch/product/USB-to-Sound-Link) but there are much cheaper options available and the shortest 3-pol cable that I found. Instead of the virtual BlackHole input and output you simply use the USB input and output.

The disadvantage could be the sound quality. There is digital to analog conversion an analog audio cable and then analog to digital conversion. This certainly doesn't improve the quality, but when I tested it for webinars the quality was more than enough. But still a lot of unnecessary conversion and this brought me to alternative 2.

Alternative 2:
A small single chip computer with good audio USB support to route the input directly to the output. There is no analog conversion involved. A device that is not that cheap but supports this in a perfect way is the Teensy 4.1. Audio USB support is included and there is almost no software development skill needed to work with it. Just download the development environment, connect the Teensy 4.1, select the correct device in the settings and enable USB Audio ports. I attached the code to download if you are interested in giving it a try. There is obviously much more possible with the Teensy such as adding a delay or sound effects.

I hope this helps until BlackHole will be again fully functioning under Monterey
[BlackHoleTeensy.ino.zip](https://github.com/ExistentialAudio/BlackHole/files/8404411/BlackHoleTeensy.ino.zip)
.
